models/issue.go

```diff
@@ -1498,3 +1507,27 @@ func updateIssue(e Engine, issue *Issue) error {
 func UpdateIssue(issue *Issue) error {
 	return updateIssue(x, issue)
 }
+
+// UpdateIssueDeadline updates an issue deadline and adds comments. Setting a deadline to 0 means deleting it.
+func UpdateIssueDeadline(issue *Issue, deadlineUnix util.TimeStamp, doer *User) (err error) {
+
+	// if the deadline hasn't changed do nothing
+	if issue.DeadlineUnix == deadlineUnix {
+		return nil
+	}
+
+	sess := x.NewSession()
+	defer sess.Close()
+
+	// Update the deadline
+	if _, err = sess.ID(issue.ID).Cols("deadline_unix").Update(&Issue{DeadlineUnix: deadlineUnix}); err != nil {
```

Comment:
We can also use `UpdateIssueCols` or `updateIssueCols` here (I did not know about that func before). It also calls issue indexer so it is more "safe" for the future changes.

`UpdateIssueCols(&Issue{ID: issue.ID, DeadlineUnix: deadlineUnix}, "deadline_unix")`

or

`updateIssueCols(sess, &Issue{ID: issue.ID, DeadlineUnix: deadlineUnix}, "deadline_unix")`

What do you think?

Reply:
Sure, I didn't knew that function either.